index.js

```diff
@@ -169,6 +172,35 @@ class K8sExecutor extends Executor {
         this.nodeSelectors = hoek.reach(options, 'kubernetes.nodeSelectors');
         this.preferredNodeSelectors = hoek.reach(options, 'kubernetes.preferredNodeSelectors');
         this.annotations = hoek.reach(options, 'kubernetes.annotations');
+        this.podRetryStrategy = (err, response, body) => {
+            const status = hoek.reach(body, 'status.phase');
+
+            return err || !status || status.toLowerCase() === 'pending';
+        };
+    }
+
+    /**
+     * Update build status message
+     * @method updateStatusMessage
+     * @param  {Object}          config                 build config of the job
+     * @param  {String}          config.apiUri          screwdriver base api uri
+     * @param  {Number}          config.buildId         build id
+     * @param  {String}          config.statusMessage   build status message
+     * @param  {String}          config.token           build temporal jwt token
+     * @return {Promise}
+     */
+    updateStatusMessage(config) {
+        const { apiUri, buildId, statusMessage, token } = config;
+        const statusMessageOptions = {
+            json: true,
+            method: 'PUT',
+            uri: `${apiUri}/v4/builds/${buildId}`,
+            body: { statusMessage },
+            headers: { Authorization: `Bearer ${token}` },
+            strictSSL: false
```

Comment:
is this missing `maxAttempts` and `retryDelay`?

Reply:
`requestretry` has a default value for those:  `maxAttempts: 5 and retryDelay: 5000`. Don't see a need to change that for this request. Not sure why we set retryDelay: 3000 in k8s-vm, probably just want it to retry faster?

Reply:
added it to make it consistent, lol